dataprofiler/profilers/profile_builder.py

```diff
@@ -2051,48 +2066,51 @@ def _update_row_statistics(
-                null_rows = null_row_indices
-                null_in_row_count = null_row_indices
-                first_col_flag = False
-            else:
-                null_rows = null_rows.intersection(null_row_indices)
-                null_in_row_count = null_in_row_count.union(null_row_indices)
+        if self.options.row_statistics.null_count.is_enabled:
+            null_rows = set()
+            null_in_row_count = set()
+            first_col_flag = True
+            for column in self._profile:
+                null_type_dict = column.null_types_index
+                null_row_indices = set()
+                if null_type_dict:
+                    null_row_indices = set.union(*null_type_dict.values())
+
+                # If sample ids provided, only consider nulls in rows that
+                # were fully sampled
+                if sample_ids is not None:
+                    # This is the amount (integer) indices were shifted by in the
+                    # event of overlap
+                    shift = column._index_shift
+                    if shift is None:
+                        # Shift is None if index is str or if no overlap detected
+                        null_row_indices = null_row_indices.intersection(
+                            data.index[sample_ids[: self._min_sampled_from_batch]]
+                        )
+                    else:
+                        # Only shift if index shift detected (must be ints)
+                        null_row_indices = null_row_indices.intersection(
+                            data.index[sample_ids[: self._min_sampled_from_batch]]
+                            + shift
+                        )
 
-        # If sample_ids provided, increment since that means only new data read
-        if sample_ids is not None:
-            self.row_has_null_count += len(null_in_row_count)
-            self.row_is_null_count += len(null_rows)
-        else:
-            self.row_has_null_count = len(null_in_row_count)
-            self.row_is_null_count = len(null_rows)
+                # Find the common null indices between the columns
+                if first_col_flag:
+                    null_rows = null_row_indices
+                    null_in_row_count = null_row_indices
+                    first_col_flag = False
+                else:
+                    null_rows = null_rows.intersection(null_row_indices)
+                    null_in_row_count = null_in_row_count.union(null_row_indices)
+
+                    # If sample_ids provided,
+                    # increment since that means only new data read
+                    if sample_ids is not None:
+                        self.row_has_null_count += len(null_in_row_count)
+                        self.row_is_null_count += len(null_rows)
+                    else:
+                        self.row_has_null_count = len(null_in_row_count)
+                        self.row_is_null_count = len(null_rows)
```

Comment:
I think we need to go the opposite direction with this. It should be outside the for loop. two tabs to the left.

Reply:
Ultimately, I suggest looking at what indent level it was prior to the change with the `if` and make sure it has not changed relative to the other code tabbing.